### ISSUE TITLE
LibriTTS vocoders 

### ISSUE THREAD

**Issue by jefflai108** [kan-bayashi/ParallelWaveGAN#340]
Hi @kan-bayashi 

A general question regarding the LibriTTS vocoders -- I am using them (melgan/hifigan/parallel_wavegan) to decode the spectrograms generated from LibriTTS Tacotron2/Transformer-TTS. When the setup was in single-speaker LJspeech, the synthesis sounds good -- the audio was crisp. However, moving to LibriTTS, the same synthesis now sounds mechanic (some of them are still decent). The attention plots from Tacotron2/Transformer-TTS are diagonal. I am guessing the issue lies in LibriTTS vocoders, as they are not as good as LJspeech vocoders? Or is the issue pertains to speaker modeling over, that the speaker encoder (speechbrain's ECAPA-TDNN) does not capture LibriTTS' speaker distribution sufficiently, which affects the synthesized spectrograms from Tacotron2/Transformer-TTS?

**Comment by kan-bayashi** [comment]
You should clarify your problem is came from text2mel part or vocoder part.

> When the setup was in single-speaker LJspeech, the synthesis sounds good -- the audio was crisp.

You mean both text2mel and vocoder are trained on LJSpeech?

> However, moving to LibriTTS, the same synthesis now sounds mechanic (some of them are still decent). 

You mean both text2mel and vocoder are trained on LibriTTS?

>  I am guessing the issue lies in LibriTTS vocoders, as they are not as good as LJspeech vocoders?

You can check LJSpeech text2mel + LibriTTS vocoder for the comparison.

**Comment by kan-bayashi** [comment]
In my experiments, multi-speaker (x-vector-based) taco 2 / transformer trained on LibriTTS is not so good.
Basically, LibriTTS includes a lot of long pause and the pause symbol is not appeared in the text.
In ESPnet recipe, I tried to remove them as possible as I can using force alignment results, but that is not perfect.
There, in terms of quality, single speaker model is definitely better than multi-speaker model when using taco2 or Transformer.

**Comment by oytunturk** [comment]
My experience with Tacotron2 so far has been that this is a very important step. Not only framing silence but also pauses between the words can cause Tacotron2 to fall apart. VAD is not a reliable way of detecting those.

**Comment by kan-bayashi** [comment]
> the synthesis depends heavily on the speaker encoder in tacotron2/transformer-TTS

Right. The quality largely depends on text2mel part. Vocoder is trained on natural speech so it is not good at recovering the missing components in generated melspectrogram. It tends to synthesis it as is.

> How important is this particular pre-processing? The silence trimming is based on force alignment to identify silence segments.

As @oytunturk, this trimming is very important especially for AR S2S models.
You can find my experiments with libritts here (https://github.com/espnet/espnet/pull/2807).
NAR models are more robust but it is definitely better to remove as possible as we can.
Most of the training problems will be solved by cleaning data.

**Comment by jefflai108** [comment]
hi @kan-bayashi 

Thanks for the response. Some updates on my side: I have identified the issue I am facing to be speaker modeling. What I did is to pick a train set speaker that sounded good, use that speaker embedding for all the test utterances, and found the test utterances sound good. I believe the issue is that there are (many) LibriTTS speakers in (both) train/test sets that the pre-trained speaker encoder does not predict well. 

What I plan to do is to take my current xvector + TTS  and finetune with one-hot speaker label. I hope this will improve speaker modeling. 

One additional question -- for LibriTTS pre-trained vocoders, are they only trained on the training set? Since LibriTTS train & test sets do not have any speaker overlap, could there be speaker modeling issue in vocoder as well?

**Comment by kan-bayashi** [comment]
Thank you for sharing your insight, @jefflai108.

> Since LibriTTS train & test sets do not have any speaker overlap, could there be speaker modeling issue in vocoder as well?

Good point. I have read some papers about the vocoder using speaker embedding to enhance the quality of unknown speaker voice. There is a room for the improvement for unknown voice. However, in my experiment, the problem is more serious in text2mel part rather than vocoder (Vocoder works well by using just multi-speaker training data without any modification of the architecture).

**Comment by jefflai108** [comment]
Thanks for your insight @kan-bayashi  
Please allow a few days for me to do more experiments.

**Comment by jefflai108** [comment]
hi @kan-bayashi 
I have some updates for the multi-speaker TTS. Some speakers are really good, and some are sub-optimal. My co-author and I are still deciding whether to submit it to Interspeech. Since you helped answer many of my questions, I would like to invite you to the author list. If you are interested in it, I can tell you more about what the submission is about and send you the demo page. Please let me know! and perhaps what's the best way to contact you?

**Comment by kan-bayashi** [comment]
@jefflai108 Thank you for sharing your status and inviting me as a co-author :)
But I have just answered a few question, I think it is not enough to be added as a co-author.
 So if you add me in acknowledgment in paper or demo hp, I would be grateful.

**Comment by jefflai108** [comment]
sure, sounds good @kan-bayashi
